cognite/neat/_graph/transformers/_prune_graph.py

```diff
@@ -1,84 +1,131 @@
 from rdflib import Graph, Namespace, URIRef
 from rdflib.query import ResultRow
-from rdflib.term import Identifier
+
+from cognite.neat._utils.rdf_ import as_neat_compliant_uri
+from cognite.neat._utils.text import sentence_or_string_to_camel
 
 from ._base import BaseTransformer
 
 
-# TODO: Handle the cse when value is None, which will not make the TextObject resolve
 class TwoHopFlattener(BaseTransformer):
     """
     Transformer that will flatten the distance between a source node, an intermediate connecting node, and a
     target property that is connected to the intermediate node.
     The transformation result is that the target property is attached directly to the source node, instead of having
     to go via the intermediate node.
-    The user can also provide a flag to decide if the intermediate node should be removed from the graph or not
-    after connecting the target property to the source node.
+    There is also an option to create a new predicate for the new direct relation, if there is a property on the
+    intermediate node that holds the value of the new predicate. If this parameter is not provided by the user, the
+    original predicate between the source node, and the resolved target value will be used.
+    The user can provide a flag to decide if the intermediate node should be removed from the graph or not
+    after connecting the target property to the source node. The default here is False.
+
+    If delete_connecting_node is not set, the expected number of triples after this transformation should be the same as
+    before the transformation.
+    If delete_connecting_node is set, the expected number of triples should be:
+        #triples_before - #destination_nodes * #destination_nodes_properties
 
         Ex. TwoHopFlattener:
 
-        Graph before flattening (with deletion of intermediate node):
-        node(A, rdf:type(Pump)) -(predicate("vendor"))>
-                                node(B, rdf:type(TextObject)) -(predicate("value"))> Literal("CompanyX")
+        Graph before transformation:
+        <http://purl.org/cognite/neat/Pump-1> a ns1:Pump ;
+        ns1:Attribute <http://purl.org/cognite/neat/Attribute-1234> .
+
+        <http://purl.org/cognite/neat/Attribute-1234> a ns1:Attribute ;
+        ns1:Name "ItemTagAssignmentClass" ;
+        ns1:Value "D-20PSV0002" .
+
+        Graph after transformation with args
+        (destination_node_type = ns1:Attribute, namespace = ns1, value_property_name="Value",
+        predicate_property_name = "Name", delete_connectiong_node=True):
 
-        Graph after flattening nodes with destination_node_type = rdf:type(TextObject), property_predicate = :value,
-        and property_name = "value":
+        <http://purl.org/cognite/neat/Pump-1> a ns1:Pump ;
+        ns1:ItemTagAssignmentClass "D-20PSV0002" .
 
-        node(A, rdf:type(Pump)) -(predicate("vendor"))> Literal("CompanyX")
+        Number of triples after operation: 5 - 1*3 = 2
 
```

Comment:
as we have been using source and target in our code base suggest renaming destination to target